### PR TITLE
CI: Uninstall nomkl & 32 bit Interval tests

### PR DESCRIPTION
**Pull request by mroeschke**

### CODE REVIEW

File: .github/actions/build_pandas/action.yml [hunk 1]
```diff
@@ -22,6 +22,13 @@ runs:
         fi
       shell: bash -el {0}
 
+    - name: Uninstall nomkl
+      run: |
+        if conda list nomkl 1>/dev/null; then
+          conda remove nomkl -yes
```
Comment (mroeschke) [inline .github/actions/build_pandas/action.yml:28]:
```suggestion
          conda remove nomkl -y
```

**Comment by jbrockmendel** [comment]
looks like the un-xfailed tests are still complaining on the pyiodide build.  that has numpy 1.26; are other builds now using newer numpys?

**Comment by mroeschke** [comment]
Yeah it looks like the closest equivalent 32 bit build is using numpy 2.1.0. I'll just xfail these tests for `WASM` for now until numpy 2 is installed

File: .github/actions/build_pandas/action.yml [hunk 2]
```diff
@@ -22,6 +22,13 @@ runs:
         fi
       shell: bash -el {0}
 
+    - name: Uninstall nomkl
+      run: |
+        if conda list nomkl | grep nomkl 1>/dev/null; then
+          conda remove nomkl -y
+        fi
+      shell: bash -el {0}
```
Comment (jorisvandenbossche) [inline .github/actions/build_pandas/action.yml:30]: @mroeschke do you remember why this was needed? (I don't see anything in the PR description or discussion)

**Pull request merged**